server.js

```diff
@@ -1,11 +1,24 @@
 var express = require("express");
 var mongoose = require("mongoose");
+var bodyParser = require("body-parser");
```

Comment:
This middle-ware is required in order to extract the `body` from the `request` object. It is `undefined` otherwise. The other option was `multer`, and it may even be what I use in the future.